tests/unit_tests/dft/source/descriptor_tests.cpp

```diff
@@ -405,6 +406,56 @@ inline void set_readonly_values(sycl::queue& sycl_queue) {
     commit_descriptor(descriptor, sycl_queue);
 }
 
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+inline void recommit_values(sycl::queue& sycl_queue) {
+    using oneapi::mkl::dft::config_param;
+    using oneapi::mkl::dft::config_value;
+    using PrecisionType =
+        typename std::conditional_t<precision == oneapi::mkl::dft::precision::SINGLE, float,
+                                    double>;
+    using value = std::variant<config_value, std::int64_t, std::int64_t*, bool, PrecisionType>;
+
+    // this will hold a param to change and the value to change it to
+    using test_params = std::pair<config_param, value>;
+
+    oneapi::mkl::dft::descriptor<precision, domain> descriptor{ default_1d_lengths };
+    EXPECT_NO_THROW(commit_descriptor(descriptor, sycl_queue));
+
+    std::array<std::int64_t, 2> strides{ 0, 1 };
+
+    std::vector<test_params> arguments{
+        // not changeable
+        // FORWARD_DOMAIN, PRECISION, DIMENSION, COMMIT_STATUS
+        std::make_pair(config_param::LENGTHS, std::int64_t{ 10 }),
+        std::make_pair(config_param::FORWARD_SCALE, PrecisionType{ 1.2 }),
+        std::make_pair(config_param::BACKWARD_SCALE, PrecisionType{ 3.4 }),
+        std::make_pair(config_param::NUMBER_OF_TRANSFORMS, std::int64_t{ 5 }),
+        std::make_pair(config_param::COMPLEX_STORAGE, config_value::COMPLEX_COMPLEX),
+        std::make_pair(config_param::REAL_STORAGE, config_value::REAL_REAL),
+        std::make_pair(config_param::CONJUGATE_EVEN_STORAGE, config_value::COMPLEX_COMPLEX),
+        std::make_pair(config_param::PLACEMENT, config_value::INPLACE),
+        std::make_pair(config_param::INPUT_STRIDES, strides.data()),
+        std::make_pair(config_param::OUTPUT_STRIDES, strides.data()),
+        std::make_pair(config_param::FWD_DISTANCE, std::int64_t{ 6 }),
+        std::make_pair(config_param::BWD_DISTANCE, std::int64_t{ 7 }),
```

Comment:
This set of new parameters is not necessarily describing a consistent in-place configuration: is that intended? If the tests are currently passing, it is not a guarantee of future successful commit for mkl backend(s) (see [MKLD-13788](https://jira.devtools.intel.com/browse/MKLD-13788)).

Reply:
I've tried to address this by making the configuration out-of-place and batching changes.

Reply:
Thank you. I believe this is fine as long as we stick to 1D configurations (we have extra restrictions for real mD cases).